apps/editor.planx.uk/src/@planx/components/shared/BaseChecklist/model.ts

```diff
@@ -159,3 +145,154 @@ export const getLayout = ({
+  name: "atLeastOneDataField",
+  test: function ({
+    fn,
+    options = [],
+    groupedOptions = [],
+  }) {
+    if (!fn) return true;
+    const allOptions = [
+      ...options,
+      ...groupedOptions.flatMap((group) => group.children),
+    ];
+    if (!allOptions) return true;
+    const optionsWithDataValues = allOptions?.filter(
+      (option) => option?.data.val,
+    );
+    if (optionsWithDataValues?.length) return true;
+    return this.createError({
+      path: "fn",
+      message: "At least one option must also set a data field",
+    });
+  },
+};
+
+const uniqueLabelsTest: TestConfig<AnyChecklist> = {
+  name: "uniqueLabels",
+  test: function ({ options }) {
+    if (!options) return true;
+    const uniqueLabels = new Set(options.map(({ data: { text } }) => text));
+    const allUnique = uniqueLabels.size === options.length;
+    if (allUnique) return true;
+    return this.createError({
+      path: "options",
+      message: "Options must have unique labels",
+    });
+  },
+};
+
+const uniqueLabelsWithinGroupsTest: TestConfig<AnyChecklist> = {
+  name: "uniqueLabelsWithinGroups",
+  test: function ({ groupedOptions }) {
+    if (!groupedOptions) return true;
+
+    for (const group of groupedOptions) {
+      if (!group.children) continue;
+
+      const uniqueLabels = new Set(
+        group.children.map(({ data: { text } }) => text),
+      );
+      const allUnique = uniqueLabels.size === group.children.length;
+
+      if (!allUnique) {
+        return this.createError({
+          path: "options",
+          message: "Options within a single group must have unique labels",
+        });
+      }
+    }
+
+    return true;
+  }
+}
+
+const uniqueGroupTitlesTest: TestConfig<AnyChecklist> = {
+  name: "uniqueGroupTitles",
+  test: function ({ groupedOptions }) {
+    if (!groupedOptions) return true;
+
+    const uniqueGroupTitles = new Set(
+      groupedOptions.map(({ title }) => title),
+    );
+
+    const allUnique = uniqueGroupTitles.size === groupedOptions.length;
+
+    if (!allUnique) {
+      return this.createError({
+        path: "options",
+        message: "Groups must have unique titles",
+      });
+    }
+
+    return true;
+  }
+};
+
+export const baseChecklistValidationSchema =
+  baseNodeDataValidationSchema.concat(
+    object({
+      description: richText(),
+      fn: string(),
+      text: string(),
+      img: string(),
+      categories: array(
+        object({
+          title: string().trim().required(),
+          count: number().required(),
+        }),
+      ),
+    })
+    // Shared BaseChecklist tests
+    // Casting is required for Yup, tests themselves are correctly typed
+    .test(onlyOneExclusiveOptionTest as TestConfig<unknown>)
+    .test(atLeastOneDataFieldTest as TestConfig<unknown>)
+    .test(uniqueLabelsTest as TestConfig<unknown>)
+    .test(uniqueLabelsWithinGroupsTest as TestConfig<unknown>)
+    .test(uniqueGroupTitlesTest as TestConfig<unknown>)
```

Comment:
I spent a while trying to get this to work without much luck - casting here is a pragmatic compromise.